firmware/user/modes/mode_ddr.c

```diff
@@ -804,10 +955,99 @@ static void ICACHE_FLASH_ATTR ddrCheckSongEnd()
             && ddr->arrowRows[2].count == 0
             && ddr->arrowRows[3].count == 0 )
     {
+        if (!ddr->didLose) 
+        {
+            ddrCheckAndSubmitScore();
+        }
         ddr->mode = DDR_SCORE;
     }
 }
 
+static void ICACHE_FLASH_ATTR ddrCheckAndSubmitScore()
+{
+    
+    int noteTotal = ddr->perfects + ddr->okays + ddr->misses;
+    if (ddr->misses == 0)
+    {
+        if (ddr->okays == 0)
+        {
+            ddr->winResult.winType = DDR_ALL_PERFECT;
+        }
+        else 
+        {
+            ddr->winResult.winType = DDR_ALL_HIT;
+        }
+    } 
+    else
+    {
+        if (ddr->perfects > MANY_PERFECT_THRESHOLD * noteTotal)
+        {
+            ddr->winResult.winType = DDR_MANY_PERFECT;
+        }
+        else if (ddr->okays + ddr->perfects > MANY_HIT_THRESHOLD * noteTotal)
+        {
+            ddr->winResult.winType = DDR_MANY_HIT;
+        }
+        else
+        {
+            ddr->winResult.winType = DDR_PASS;
+        }
+    }
+
+    ddr->winResult.score = SCORE_PERFECT * ddr->perfects + SCORE_HIT * ddr->okays + SCORE_MISS * ddr->misses;
+
+    ddrWinResult_t* currentDiffScores;
+
+    switch(ddr->currentDifficultyType)
+    {
+        default:
+        case DDR_HARD:
+            currentDiffScores = ddr->highScores.hardWins;
+            break;
+        case DDR_MEDIUM:
+            currentDiffScores = ddr->highScores.mediumWins;
+            break;
+        case DDR_EASY:
+            currentDiffScores = ddr->highScores.easyWins;
+            break;
+    }
+
+    ddr->isNewHighScore = false;
+    int replaceIndex;
```

Comment:
I may be reading this incorrectly, but wouldn't you want to do an insertion, not a replacement, in order to keep all the high scores sorted?

Reply:
It was a horribly implemented insertion. 😆 
I've rewritten it now